proxy/http/HttpTransact.cc

```diff
@@ -1546,6 +1546,30 @@ HttpTransact::OSDNSLookup(State *s)
     TRANSACT_RETURN(SM_ACTION_SEND_ERROR_CACHE_NOOP, nullptr);
   }
 
+  if (s->redirect_info.redirect_in_process) {
+    if (HttpConfigParams::REDIRECT_ENABLED_FOLLOW == s->http_config_param->redirect_enabled) {
+      TxnDebug("http_trans", "[OSDNSLookup] Invalid redirect address. Following");
```

Comment:
Since we have 5 separate checks to do when not configured to follow, I am handling this case separately.